bundle/Cache/PersistenceCachePurger.php

```diff
@@ -119,25 +120,37 @@ public function content($locationIds = null)
             $locationIds = array($locationIds);
         }
 
+        if ($contentIds === null) {
+            $contentIds = array();
+        } elseif (!is_array($contentIds)) {
+            $contentIds = array($contentIds);
+        }
+
         foreach ($locationIds as $id) {
             if (!is_scalar($id)) {
                 throw new InvalidArgumentType('$id', 'int[]|null', $id);
             }
 
             try {
-                $location = $this->locationHandler->load($id);
-                $this->cache->clear('content', $location->contentId);
-                $this->cache->clear('content', 'info', $location->contentId);
-                $this->cache->clear('content', 'info', 'remoteId');
-                $this->cache->clear('content', 'locations', $location->contentId);
-                $this->cache->clear('user', 'role', 'assignments', 'byGroup', $location->contentId);
-                $this->cache->clear('user', 'role', 'assignments', 'byGroup', 'inherited', $location->contentId);
+                $contentIds[] = $this->locationHandler->load($id)->contentId;
             } catch (NotFoundException $e) {
                 $this->logger->notice(
                     "Unable to load the location with the id '$id' to clear its cache"
                 );
             }
         }
+        foreach ($contentIds as $id) {
+            if (!is_scalar($id)) {
+                throw new InvalidArgumentType('$id', 'int[]|null', $id);
+            }
+
+            $this->cache->clear('content', $id);
+            $this->cache->clear('content', 'info', $id);
+            $this->cache->clear('content', 'info', 'remoteId');
```

Comment:
can move this down into related cache while at it, does not need to be in the loop

Reply:
and what about move these 6 calls to a dedicated method? (maybe protected)

Reply:
nah, they are not relevant for master